src/Build/Evaluation/Evaluator.cs

```diff
@@ -1840,9 +1843,16 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
                     }
                 }
 
+                IDisposable assemblyLoadTracker = null;
                 // Combine SDK path with the "project" relative path
                 try
                 {
+                    // Is the sdk resolver a custom type?
```

Comment:
Aren't SDK resolvers in different assemblies? We have our built-in one, but NuGet owns theirs, C++ owns theirs, etc.

Reply:
That's a good point - I believe that if it's not built-in, than the default tracking has a sense, as it might not be obvious from where it was loaded. (similarly to this comment about roslyn tasks: https://github.com/dotnet/msbuild/pull/8316#discussion_r1103814426). 
What do you think?

Btw. this code was accidentaly performin the filtering twice - I removed the superfluous call wrapping the `StartTracking`, as unified (and bit more comprehensive) filtering is performed within `StartTracking`.
It filters as well based on the typename - https://github.com/dotnet/msbuild/pull/8316/files/f6c696399c13667bc7742a8e94c0cfc3c79e46aa#diff-7f217b4ccb76592a2ab78dd7f8332509dfa2e2216d8a6032f956cbdf48df8d03R86-R96

Reply:
There is a  opt-in trait (via `MSBUILDLOGALLASSEMBLYLOADS` env var) to stop any filtering.
So if for any reason any component of interest would not be hooked (e.g. due to being placed to 'Microsoft.Build' namespace), it can allways be opted-in.